x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -30,9 +31,14 @@ export interface AsyncSearchResponse<T> {
   response: SearchResponse<T>;
 }
 
+export function isAsyncSearchResponse<T>(response: any): response is AsyncSearchResponse<T> {
```

Comment:
(a) instead of `any` you could use the response base class.
(b) no need to export this function as well as `AsyncSearchResponse` - it's not used (and shouldn't be used) externally.

Reply:
👍 I'll give this a try